src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1705,6 +1710,28 @@ public function isXmlHttpRequest()
         return 'XMLHttpRequest' == $this->headers->get('X-Requested-With');
     }
 
+    /**
+     * Checks whether the client browser prefers safe content or not according to RFC8674.
+     *
+     * @see https://tools.ietf.org/html/rfc8674
+     */
+    public function preferSafeContent(): bool
+    {
+        if (null !== $this->isSafeContentPreferred) {
+            return $this->isSafeContentPreferred;
+        }
+
+        if (!$this->isSecure()) {
+            $this->isSafeContentPreferred = false;
```

Comment:
maybe add `// see https://tools.ietf.org/html/rfc8674#section-3`? I hade to open the RFC to understand this "if"